torch/csrc/autograd/engine.cpp

```diff
@@ -398,6 +398,65 @@ void add_node_to_current_graph_task_exec_info(Node* fn) {
   current_graph_task->exec_info_[fn].needed_ = true;
 }
 
+// NB: The engine itself does not use the outputs of this function.
+//
+//     WARNING: the predicted order is only correct when the following
```

Comment:
Should we use the new single threaded backward to enforce this?

Reply:
Whoops was thinking of something different - yes, you're right, it's the same